mdast_cli/distribution_systems/firebase.py

```diff
@@ -0,0 +1,92 @@
+import os
+import sys
+from hashlib import sha1
+from time import time
+
+import requests
+
+try:
+    from ..helpers.logging import Log
+    from .base import DistributionSystem
+except ImportError:
+    from mdast_cli.distribution_systems.base import DistributionSystem
+    from mdast_cli.helpers.logging import Log
+
+
+class Firebase(DistributionSystem):
+    """
+    Downloading application from Firebase
+    """
+    download_path = 'downloaded_apps'
+    url = 'https://console.firebase.google.com'
+
+    def __init__(self, project_id, app_id, app_code, api_key, SID, HSID, SSID, APISID, SAPISID, file_name=None):
+        super().__init__(app_id, app_code)
+
+        self.project_id = project_id
+        self.app_id = app_id
+        self.app_code = app_code
+        self.api_key = api_key
+        self.SID = SID
+        self.HSID = HSID
+        self.SSID = SSID
+        self.APISID = APISID
+        self.SAPISID = SAPISID
+        self.file_name = file_name
+
+    def calculate_sapisid_hash(self):
+        """Calculates SAPISIDHASH based on cookies. Required in authorization to download apk from firebase"""
+        epoch = int(time())
+        sha_str = ' '.join([str(int(epoch)), self.SAPISID, self.url])
+        sha = sha1(sha_str.encode())
+        return f'SAPISIDHASH {int(epoch)}_{sha.hexdigest()}'
+
+    def download_app(self):
+        SAPISIDHASH = Firebase.calculate_sapisid_hash(self)
```

Comment:
Why you calling Firebase again? 
Just `self.calculate_sapisid_hash()` will be enough